src/harness/testcases/WINNF_FT_S_PCR_testcase.py

```diff
@@ -0,0 +1,643 @@
+
+    # At least one of the CBSDs is close to the boundary of the Service Area,
+    # so that -96 dBm/10 MHz protection contour extends beyond the service area boundary.
+    device_b['installationParam']['latitude'], device_b['installationParam'][
+        'longitude'] = 38.70645, -100.46034
+
+    # placed the device_c in between device_a and device_b within service area.
+    device_c['installationParam']['latitude'], device_c['installationParam'][
+        'longitude'] = 38.72281, -100.50103
+
+    # Set the AntennaGain and EIRP capability.
+    device_a['installationParam']['eirpCapability'] = 30
+    device_b['installationParam']['eirpCapability'] = 47
+    device_a['installationParam']['antennaGain'] = 16
+    device_b['installationParam']['antennaGain'] = 16
+
+    conditionals_b = {
+        'cbsdCategory': device_b['cbsdCategory'],
+        'fccId': device_b['fccId'],
+        'cbsdSerialNumber': device_b['cbsdSerialNumber'],
+        'airInterface': device_b['airInterface'],
+        'installationParam': device_b['installationParam'],
+        'measCapability': device_b['measCapability']
+    }
+    conditionals = [conditionals_b]
+    del device_b['installationParam']
+    del device_b['cbsdCategory']
+    del device_b['airInterface']
+    del device_b['measCapability']
+
+    # Create the actual config.
+    devices = [device_a, device_b, device_c]
+    config = {
+        'registrationRequests': devices,
+        'conditionalRegistrationData': conditionals,
+        'palRecords': pal_records,
+        'sasTestHarnessCert': os.path.join('certs', 'sas.cert'),
+        'sasTestHarnessKey': os.path.join('certs', 'sas.key')
+    }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_PCR_2_default_config)
+  def test_WINNF_FT_S_PCR_2(self, config_filename):
+    """Successful Maximum PPA Boundary Creation Clipped by Service Area Boundary.
+
+    Checks the maximum PPA boundary, being clipped by the Service Area composed of
+    one or more adjacent Census Tracts.
+    Checks PPA generated by SAS UUT shall be fully contained within the service area.
+    """
+    # Load the Config file.
+    config = loadConfig(config_filename)
+
+    # Inject the PAL records.
+    for pal_record in config['palRecords']:
+      self._sas_admin.InjectPalDatabaseRecord(pal_record)
+
+    # Register devices and  check response.
+    cbsd_ids = self.assertRegistered(config['registrationRequests'],
+                                     config['conditionalRegistrationData'])
+
+    # Trigger SAS UUT to create a PPA boundary.
+    pal_ids = [record['palId'] for record in config['palRecords']]
+    ppa_creation_request = {
+        "cbsdIds": cbsd_ids,
+        "palIds": pal_ids
+    }
+
+    # Trigger PPA Creation to SAS UUT.
+    ppa_id = self.triggerPpaCreationAndWaitUntilComplete(ppa_creation_request)
+    logging.debug('ppa_id received from SAS UUT:%s', ppa_id)
+
+    # Notify SAS UUT about SAS Harness and trigger Full Activity Dump and
```

Comment:
You haven't done the first part of this (notify).

Reply:
Notify   SAS UUT via injectPeer SAS admin API is already done inside the   method     triggerFadGenerationAndRetrievePpaZone . So avoid the confusion we have   repharse the     comment line accordingly